lib/rubocop_director/rubocop_stats.rb

```diff
@@ -7,22 +7,48 @@
 module RubocopDirector
   class RubocopStats
     include Dry::Monads[:result]
+    include Dry::Monads::Do.for(:fetch)
+
+    TEMP_CONFIG_PATH = "./.temp_rubocop.yml"
 
     def fetch
-      _, stderr = Open3.capture3("sed '/todo/d' ./.rubocop.yml > tmpfile; mv tmpfile ./.rubocop.yml")
-      if stderr.length > 0
-        return Failure("Failed to remove TODO from rubocop config: #{stderr}")
-      end
+      config = yield load_config
+      yield temp_config(initial_config: config)
+
+      stats = yield generate_stats
+      yield remove_temp_config
+
+      Success(stats)
+    end
+
+    private
 
-      stdout, stderr = Open3.capture3("bundle exec rubocop --format json")
+    def load_config
+      Success(YAML.load_file("./.rubocop.yml"))
+    rescue
```

Comment:
A better practice to have an explicit list of errors to be rescued and insert the error message to the `Failure` as well — otherwise people won't be able to report the bug properly

Reply:
got it, thanks!